### PR TITLE
EG-464 Corda returns incorrect exit code in case if node is started with unknown/missing option

### PR DESCRIPTION
**Pull request by pnemeth**
Use a custom error handler to handle ParseException and ExecutionException and return error code 1

Added RunCordaNodeReturnCodeTests to check error codes for various arguments

### CODE REVIEW

**Review by JamesHR3 — COMMENTED** [review]
This seems reasonable, but I think it's probably worth verifying that for the cases where `verbose` did work before this change (so any of the tools) that we'll still print `Throwable` stack traces if `verbose` is set. Have you done that already?

**Comment by pnemeth** [comment]
need to fix a bug

**Comment by pnemeth** [comment]
reopening for review

**Comment by pnemeth** [comment]
> This seems reasonable, but I think it's probably worth verifying that for the cases where `verbose` did work before this change (so any of the tools) that we'll still print `Throwable` stack traces if `verbose` is set. Have you done that already?

Cheers, retested this after latest commit, and it works fine.

**Review by JamesHR3 — DISMISSED** [review]
LGTM!

**Comment by woggioni** [comment]
it would probably be good to have a unit test that checks that `--verbose` prints the stacktrace while the default is to only print the exception message..

I mean only testing `CordaCliWrapper`, like implementing a simple cli in the unit test with a single command that always throws an exception.

This would set the expected behavior of this class, preventing other people to change it again and also will make it clear that, if that feature doesn't work in any of the implementing cli, it is because that is broken (swallowing and managing exception itself), not this class

**Comment by pnemeth** [comment]
> it would probably be good to have a unit test that checks that `--verbose` prints the stacktrace while the default is to only print the exception message..
> 
> I mean only testing `CordaCliWrapper`, like implementing a simple cli in the unit test with a single command that always throws an exception.
> 
> This would set the expected behavior of this class, preventing other people to change it again and also will make it clear that, if that feature doesn't work in any of the implementing cli, it is because that is broken (swallowing and managing exception itself), not this class

Cheers Walter, good idea, looking into this.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/SampleCordaCliWrapper.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    companion object {
+        @JvmStatic
+        fun main(args: Array<String>) {
+            SampleCordaCliWrapper().start(args)
+        }
+    }
+
+    @CommandLine.Option(names = ["--sample-command"],
+            description = [ "Root directory containing the node configuration files and CorDapp JARs that will form the test network.",
```
Comment (JamesHR3) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/SampleCordaCliWrapper.kt]: Can we change this description to better match what the test app does?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/ProcessUtilities.kt [hunk 2]
```diff
@@ -37,7 +37,9 @@ object ProcessUtilities {
             extraJvmArguments: List<String> = emptyList(),
             maximumHeapSize: String? = null,
             identifier: String = "",
-            environmentVariables: Map<String,String> = emptyMap()
+            environmentVariables: Map<String,String> = emptyMap(),
+            inheritIO: Boolean = true
+
```
Comment (JamesHR3) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/ProcessUtilities.kt]: Remove the space here

**Review by JamesHR3 — CHANGES_REQUESTED** [review]
Couple of really small changes.

How confident are we that the test will continue to pass if e.g. the stack trace of the exception were to change? This could happen if the code in `cliutils` were changed. (From my reading of it I think we're ok, but always worth verifying regexs I think.)

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/SampleCordaCliWrapper.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package net.corda.cliutils
```
Comment (woggioni) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/SampleCordaCliWrapper.kt]: I believe this file should go in the `integration-test` folder...

**Comment by pnemeth** [comment]
> Couple of really small changes.
> 
> How confident are we that the test will continue to pass if e.g. the stack trace of the exception were to change? This could happen if the code in `cliutils` were changed. (From my reading of it I think we're ok, but always worth verifying regexs I think.)

I think the regexp is quite general, but of course if it changes a lot we will have to update the test.

**Review by JamesHR3 — APPROVED** [review]
LGTM

**Review by roastario — APPROVED** [review]

**Pull request merged**